__tests__/components/AddressBarInput.spec.js

```diff
@@ -67,4 +64,112 @@ describe( 'AddressBarInput', () =>
+                </Provider > );
+            instance = wrapper.instance();
+        } );
+
+        afterEach( () =>
+        {
+            wrapper.unmount();
+        } );
+
+        it( 'check on onBlur,handleBlur is called', () =>
+        {
+            const handleBlur = jest.fn();
+            wrapper = mount(
+                <Provider store={ store } >
+                    <AddressBarInput { ...props } onBlur={ handleBlur } />
+                </Provider > );
+            const input = wrapper.find( 'Input' );
+            input.simulate( 'blur' );
+            expect( handleBlur ).toHaveBeenCalled();
+        } );
+
+        it( 'check on onBlur,onBlur() is called', () =>
+        {
+            const input = wrapper.find( 'Input' );
+            input.simulate( 'blur' );
+            expect( props.onBlur ).toHaveBeenCalled();
+        } );
+
+        it( 'check on onFocus,handleFocus is called', () =>
+        {
+            const handleFocus = jest.fn();
+            wrapper = mount(
+                <Provider store={ store } >
+                    <AddressBarInput { ...props } onFocus={ handleFocus } />
+                </Provider > );
+            instance = wrapper.instance();
+            const input = wrapper.find( 'Input' );
+            input.simulate( 'focus' );
+            expect( handleFocus ).toHaveBeenCalled();
+        } );
+
+        it( 'check on onFocus,onFocus() is called', () =>
+        {   
+            wrapper = mount(
+                <Provider store={ store } >
+                    <AddressBarInput { ...props } />
+                </Provider > );
+            instance = wrapper.instance();
+            const input = wrapper.find( 'Input' );
+            input.simulate( 'focus' );
+            expect( props.onFocus ).toHaveBeenCalled();
+        } );
+
+        it( 'check if onKeyPress triggers', () =>
+        {
+            const input = wrapper.find( 'Input' );
+            expect( input.simulate( 'keyPress', { key: 'Enter', keyCode: 13, which: 13 } ) ).toBeTruthy();
+        } );
+
+        it( 'check on onKeyPress if activeTab is called', () =>
+        {
+            const input = wrapper.find( 'Input' );
+            input.simulate( 'keyPress', { key: 'Enter', keyCode: 13, which: 13 } );
+            expect( props.updateActiveTab ).toHaveBeenCalled();
+        } );
+
+        it( 'check if onChange triggers', () =>
+        {
+            const input = wrapper.find( 'Input' );
+            input.value = '123456';
+            input.simulate( 'change' );
+            expect( input.simulate( 'change' ) ).toBeTruthy();
+        } );
+
+        it( 'check on onChange, if onSelect() is called', () =>
+        {
+            const input = wrapper.find( 'Input' );
+            input.value = '123456';
+            input.simulate( 'change' );
+            expect( props.onSelect ).toHaveBeenCalled();
+        } );
+    } );
+    describe( 'componentWillReceiveProps( nextProps )', () =>
```

Comment:
I don't think we need to check this as it's internal React functionality.

We can assume that's well tested.

Reply:
alright then. I'll drop this from the tests.